crates/ruff_python_formatter/src/verbatim.rs

```diff
@@ -573,33 +596,275 @@ impl Format<PyFormatContext<'_>> for TrailingFormatOffComment<'_> {
     }
 }
 
-struct VerbatimText(TextRange);
+/// Stores the indentation of a statement by storing the number of indentation characters.
+/// Storing the number of indentation characters is sufficient because:
+/// * Two indentations are equal if they result in the same column, regardless of the used tab size.
+///   This implementation makes use of this fact and assumes a tab size of 1.
+/// * The source document is correctly indented because it is valid Python code (or the formatter would have failed parsing the code).
+#[derive(Copy, Clone)]
+struct Indentation(u32);
+
+impl Indentation {
+    fn from_stmt(stmt: &Stmt, source: &str) -> Indentation {
+        let line_start = Locator::new(source).line_start(stmt.start());
+
+        let mut indentation = 0u32;
+        for c in source[TextRange::new(line_start, stmt.start())].chars() {
+            if matches!(c, ' ' | '\t') {
```

Comment:
you taught me about `is_python_whitespace`

Reply:
This is intentionally different because we don't want to trim any form-feed characters (see the `form_feed.py` test case)

Reply:
could you add a comment about this?